orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/RegionCollectorSpec.groovy

```diff
@@ -0,0 +1,107 @@
+package com.netflix.spinnaker.orca.pipeline.util
+
+import spock.lang.Specification
+
+import static com.netflix.spinnaker.orca.test.model.ExecutionBuilder.pipeline
+import static com.netflix.spinnaker.orca.test.model.ExecutionBuilder.stage
+
+class RegionCollectorSpec extends Specification {
+  RegionCollector regionCollector
+
+  def setup() {
+    regionCollector = new RegionCollector()
+  }
+
+  def "should find all regions from subsequent deploys"() {
+    when:
+    def regionSet = regionCollector.getRegionsFromDescendentStages(execution.stageByRef("1"))
+
+    then:
+    regionSet == allDeployRegions
+
+    where:
+    deployClusters = [
+      [
+        cloudProvider: "aws",
+        availabilityZones: [ region1: [], region2: []]
+      ],
+      [
+        cloudProvider: "aws",
+        availabilityZones: [ region3: [] ]
+      ]
+    ]
+
+    allDeployRegions = deployClusters.sum {it->it.availabilityZones.keySet()}
+
+    execution = pipeline {
+      stage {
+        name = "findImage"
+        type = "findImage"
+        refId = "1"
+        context = [ cloudProvider: "aws"]
+      }
+
+      stage {
+        name = "deploy"
+        type = "deploy"
+        context = [
+          clusters: deployClusters
+        ]
+        requisiteStageRefIds = ["1"]
+      }
+    }
+  }
+
+  def "should find all regions from subsequent canary deploys"() {
+    when:
+    def regionSet = regionCollector.getRegionsFromDescendentStages(execution.stageByRef("1"))
+
+    then:
+    regionSet == allDeployRegions
+
+    where:
+    deployCluster1 = [
```

Comment:
semantics ... but most of what you've got in the `where:` block is stuff I would put in a `given:` block

Reply:
yeah, you're right... fixing